internal/pkg/discovery/registry/kubernetes.go

```diff
@@ -94,7 +94,7 @@ func AffiliateFromNode(node *v1.Node) *cluster.AffiliateSpec {
 	}
 
 	// Machine type is derived from node roles.
-	_, labelMaster := node.Labels[constants.LabelNodeRoleMaster]
+	_, labelMaster := node.Labels[constants.LabelNodeRoleControlPlane]
```

Comment:
I guess this should stay for backwards compatibility? or it should be removed, as control plane label is checked below

Reply:
i think this one is fine, since on upgrade it would get a new label and will still work

Reply:
oh I mean that if you look on lines 97-98 and below what we do is we check for either `master` or `control-plane` labels, so check for `master` label can be dropped

Reply:
ahh yes, saw that while fixing other issues.